lua/ttt2/libraries/targetid.lua

```diff
@@ -473,9 +473,9 @@ function targetid.HUDDrawTargetIDRagdolls(tData)
 
 	local corpse_found = CORPSE.GetFound(ent, false) or not DetectiveMode()
 	local corpse_ply = corpse_found and CORPSE.GetPlayer(ent) or false
-	local role_found = (corpse_found and ent.bodySearchResult and ent.bodySearchResult.subrole) or (corpse_ply and corpse_ply:GetSubRole())
 	local binoculars_useable = IsValid(c_wep) and c_wep:GetClass() == "weapon_ttt_binoculars" or false
-	local roleData = (corpse_ply and corpse_ply:GetSubRoleData()) or roles.GetByIndex(role_found and ent.bodySearchResult.subrole or ROLE_INNOCENT)
+	local role_found = (corpse_found and ent.bodySearchResult and ent.bodySearchResult.subrole) or (corpse_ply and IsValid(corpse_ply) and corpse_ply:GetSubRole())
+	local roleData = (corpse_ply and IsValid(corpse_ply) and corpse_ply:GetSubRoleData()) or roles.GetByIndex(role_found and ent.bodySearchResult.subrole or ROLE_INNOCENT)
```

Comment:
I'm wondering if ROLE_INNOCENT is the right choice here. Maybe ROLE_NONE would be better? I'm asking because we used innocent as unknown role in the early days before we had the none role

Reply:
afaik i've never actually triggered this part of the code but i'll test it out 
![image](https://github.com/TTT-2/TTT2/assets/5711436/7628fe1f-2524-4005-8a52-884af2006058)
the completely blank role icon is a little jarring if this condition DOES ever get tripped

Reply:
understandable, thanks

Reply:
i think if the icon for `ROLE_NONE` were similar to the `NO TEAM` icon in this context it'd be a little more suitable
![image](https://github.com/TTT-2/TTT2/assets/5711436/018df159-19d4-469e-be9e-57fcd0705b94)